ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeOption.java

```diff
@@ -38,19 +38,25 @@
    *
    * @see SerdeOption#WRAP_SINGLE_VALUES
    */
-  UNWRAP_SINGLE_VALUES(SerdeFeature.UNWRAP_SINGLES),
+  UNWRAP_SINGLE_VALUES(SerdeFeature.UNWRAP_SINGLES, false),
 
   /**
    * Key version of {@link #UNWRAP_SINGLE_VALUES}.
    */
-  UNWRAP_SINGLE_KEYS(SerdeFeature.UNWRAP_SINGLES);
+  UNWRAP_SINGLE_KEYS(SerdeFeature.UNWRAP_SINGLES, true);
 
+  private final boolean isKey;
   private final SerdeFeature requiredFeature;
 
-  SerdeOption(final SerdeFeature requiredFeature) {
+  SerdeOption(final SerdeFeature requiredFeature, final boolean isKey) {
```

Comment:
Does it make sense to assume all serde options will always be either key- or value-specific, never both nor neither? Can't think of why this wouldn't be true in the future but seemed worth double-checking.

Reply:
That's the assumption at the moment.  If that changes... we change the code! ;)

Ideally, we'd ditch `SerdeOption` and just keep `SerdeFeature`, but that's a bigger change.